pkg/debug/spec.sh

```diff
@@ -324,11 +267,129 @@ __EOT__
+    # no kernel module -> pillar does not get confused -> passthrough of the device is okay
+    for i in /sys/class/net/*
+    do
+        i=$(basename "$i")
+        local devicepath
+        devicepath=$(realpath "/sys/class/net/$i/../../")
+        local isUSB
+        isUSB=$(echo "$devicepath" | grep -Eo '/usb[0-9]/' || true)
+
+        if [ "$isUSB" = "" ];
+        then
+            continue
+        fi
+        local netdevpaths="$netdevpaths $devicepath"
+    done
+
+    for i in $(find /sys/devices/ -name uevent | grep -E '/usb[0-9]/' | grep -E '/[0-9]-[0-9](\.[0-9]+)?/uevent')
+    do
+        local ignore_dev=0
+        local devicepath
+        devicepath=$(dirname "$i")
+        local busAndPort
+        busAndPort=$(echo "$i" | grep -Eo '/[0-9]-[0-9](\.[0-9]+)?/uevent' | grep -Eo '[0-9]-[0-9](\.[0-9]+)?')
+        local assigngrp
+        assigngrp="USB${busAndPort}"
+
+        local bus
+        bus=$(echo "${busAndPort}" | cut -d - -f1)
+        local port
+        port=$(echo "${busAndPort}" | cut -d - -f2)
+        local usbaddr="$bus:$port"
+
+        local ifname
+        ifname="$(ls -1 "${devicepath}/*/net" 2>/dev/null || true)"
+        local cost=0
+        local isModem
+        isModem=$(grep -Eo 'cdc_mbim|qmi_wwan' "${devicepath}"/*/uevent 2>/dev/null)
+        if [ "${isModem}" != "" ]
+        then
+            cost=10
+            assigngrp="modem${busAndPort}"
+            assigngrp=$(get_assignmentgroup "${ifname}" "${pciaddr}")
+        fi
+
+        local pciaddr
+        pciaddr=$(echo "$i" | grep -Eo '/pci[^/]+/[^/]+/usb' | cut -d / -f3)
+        local parentassigngrp
+        parentassigngrp="group$(pci_iommu_group "${pciaddr}")"
+        if [ "$parentassigngrp" = "group" ]
+        then
+            parentassigngrp=""
+        fi
+        local product
+        product=$(cat "$devicepath/product" 2> /dev/null || true)
+        local label
+        label=$(echo "$product" | tr -cd '[:alnum:]._-')
+        if [ "$label" = "" ]
+        then
+            label="USB${busAndPort}"
+        fi
+
+        type=$(grep -Eo '^TYPE=[0-9]+/' "$i"| grep -Eo '[0-9]+')
+        # ignore USB hubs
+        if [ "$type" = "9" ];
+        then
+            ignore_dev=1
+        fi
+
+        # ignore network cards but allow usb modems
+        if [ "${isModem}" = "" ]
+        then
+            for netdevpath in $netdevpaths
+            do
+                # check if devicepath starts with netdevpath
+                case $netdevpath in "$devicepath"*)
+                    ignore_dev=1
+                esac
+            done
+        fi
+
+        if [ "$ignore_dev" = "1" ]
+        then
+            continue
+        fi
+
+        cat <<__EOT__
+    {
+      "ztype": "IO_TYPE_UNSPECIFIED",
+      "phylabel": "${label}",
+      "assigngrp": "${assigngrp}",
+      "cost": ${cost},
+      "phyaddrs": {
+        "usbaddr": "$usbaddr"
+      },
+      "logicallabel": "$label",
+      "usagePolicy": {}
+    },
```

Comment:
What about adding VendorID:ProductID as `description` when verbose is enabled? Although not recognized and not allowed by the controller, it can be used by the user to determine which device each entry represents. That was actually the reason why AT commands were used to get some info for each modem. With multiple modems, it can be difficult to tell them apart and address of modem attachment is not something that users can easily map to hardware. (modem IMEI would be ideal but it is difficult to get from a simple shell script)

Reply:
As I see it, `picocom` was used to get the description of the device, like product name.

Now the product name is set in the `phylabel`, e.g.:
```
    {
      "ztype": "IO_TYPE_WWAN",
      "phylabel": "SierraWirelessEM7565QualcommSnapdragonX16LTE-A",
      "assigngrp": "",
      "cost": 10,
      "phyaddrs": {
        "usbaddr": "3:1.4"
      },
      "logicallabel": "SierraWirelessEM7565QualcommSnapdragonX16LTE-A",
      "usagePolicy": {}
    },
```

https://github.com/lf-edge/eve/pull/3555/files#diff-5082c5ce36254a19dd88fff26f3d05bc815ca75f4efb1a8b0d4868803ac6da51R335 <- that's how it gets the product string
But still, if we have the same modem twice, the `phylabel` would be indistinguishable, so I will add the bus and port number, so it will look like: `SierraWirelessEM7565QualcommSnapdragonX16LTE-A-USB3-1.4`

Reply:
Note that logicallabel is used by the controller as the adapter name and it will appear in UI. So for that we should generate something shorter for nicer visualization, like modem0, modem1, etc. I think phylabel can be long like that (but we should check with the controller team if there are any string length restrictions imposed in the database).

Reply:
But I also think that using raw and shorter "VendorID:ProductID" (+ address) for physical label would be fine as well.

Reply:
I though `phylabel` and `logicallabel` have to be the same string - I ran into issues with them being different.

I would actually prefer seeing the name of the modem over just `modem0`.

Reply:
It is expected that the user will edit the generated device model anyway and put there some label relevant to their deployment. Logical label should really be something short that can be displayed in UI without taking half the screen.
Physical label can be different, at least in newer EVE versions it is not a problem anymore.